Sources/ConcordiumSwiftSDK/Model/Account.swift

```diff
@@ -16,9 +49,373 @@ struct AccountAddress {
         self.bytes = bytes // excludes initial version byte
     }
 }
+
 typealias SequenceNumber = UInt64
 
 struct NextAccountSequenceNumber {
     let sequenceNumber: SequenceNumber
     let allFinal: Bool
 }
+
+typealias AccountIndex = UInt64
+typealias CredentialRegistrationId = Data // 48 bytes
+typealias Amount = UInt64
+typealias CredentialIndex = UInt32
+typealias ArCurve = Data
+typealias AttributeKind = Data
+typealias AccountThreshold = UInt32
+typealias AccountEncryptionKey = String
+typealias PublicKey = Data
+
+func dateFromUnixTimeMillis(_ timestamp: UInt64) -> Date {
+    Date(timeIntervalSince1970: TimeInterval(timestamp) / 1000)
+}
+
+struct Release {
+    let timestamp: Date
+    let amount: Amount
+    let transactions: [TransactionHash]
+
+    static func fromGrpcType(_ grpc: Concordium_V2_Release) -> Release {
+        .init(
+                timestamp: dateFromUnixTimeMillis(grpc.timestamp.value),
+                amount: grpc.amount.value,
+                transactions: grpc.transactions.map {
+                    $0.value
+                }
+        )
+    }
+}
+
+struct AccountReleaseSchedule {
+    let total: Amount
+    let schedule: [Release]
+
+    static func fromGrpcType(_ grpc: Concordium_V2_ReleaseSchedule) -> AccountReleaseSchedule {
+        .init(total: grpc.total.value, schedule: grpc.schedules.map {
+            .fromGrpcType($0)
+        })
+    }
+}
+
+struct Versioned<V> {
+    let version: UInt32
+    let value: V
+}
+
+typealias KeyIndex = UInt32
+typealias VerifyKey = PublicKey
+typealias SignatureThreshold = UInt32
+typealias IpIdentity = UInt32
+typealias AttributeTag = UInt32
+typealias Threshold = UInt32
+typealias ArIdentity = UInt32
+
+extension VerifyKey {
+    static func fromGrpcType(_ grpc: Concordium_V2_AccountVerifyKey) -> VerifyKey? {
+        switch grpc.key {
+        case nil:
+            return nil
+        case let .ed25519Key(d):
+            return d
+        }
+    }
+}
+
+struct CredentialPublicKeys {
+    let keys: [KeyIndex: VerifyKey]
+    let threshold: SignatureThreshold
+
+    static func fromGrpcType(_ grpc: Concordium_V2_CredentialPublicKeys) throws -> CredentialPublicKeys {
+        CredentialPublicKeys(
+                keys: try grpc.keys.mapValues {
+                    try VerifyKey.fromGrpcType($0) ?? {
+                        throw GrpcError.requiredValueMissing("credential public keys")
+                    }()
+                },
+                threshold: grpc.threshold.value
+        )
+    }
+}
+
+struct YearMonth {
+    let year: UInt32
```

Comment:
On a closer look it seems like the Rust SDK intentially reduces the type size of some values that are defined as 32 bit in the protobuf spec. @abizjak can you shed some light on this?

Reply:
The real type is this https://github.com/Concordium/concordium-base/blob/main/rust-src/concordium_base/src/id/types.rs#L379

Protobuf does not have u16 and u8 so we do what we can use there. But the values will always fit the range stated there.

Reply:
Further to this, the value for year is actually meant to be in rane 1000..=9999 and month in the obvious range 1..=12.

Also note that a JSON format is based on a string in the format YYYYMM (this will probably become relevant when you do FFI).

Reply:
I think for the purpose of this SDK it's fine to keep the types as u32. I think that's what the Java SDK does as well. Do you agree @abizjak ? The ranges are already documented in the docstring. I'll add a method for the JSON format.

Reply:
Yes.

Reply:
done